src/jdk.compiler/share/classes/com/sun/tools/javac/comp/Attr.java

```diff
@@ -5403,125 +5403,125 @@ void attribClass(ClassSymbol c) throws CompletionFailure {
             Env<AttrContext> env = typeEnvs.get(c);
 
-            if (c.isSealed() &&
-                    !c.isEnum() &&
-                    !c.isPermittedExplicit &&
-                    c.permitted.isEmpty()) {
-                log.error(TreeInfo.diagnosticPositionFor(c, env.tree), Errors.SealedClassMustHaveSubclasses);
-            }
-
-            if (c.isSealed()) {
-                Set<Symbol> permittedTypes = new HashSet<>();
-                boolean sealedInUnnamed = c.packge().modle == syms.unnamedModule || c.packge().modle == syms.noModule;
-                for (Symbol subTypeSym : c.permitted) {
-                    boolean isTypeVar = false;
-                    if (subTypeSym.type.getTag() == TYPEVAR) {
-                        isTypeVar = true; //error recovery
-                        log.error(TreeInfo.diagnosticPositionFor(subTypeSym, env.tree),
-                                Errors.InvalidPermitsClause(Fragments.IsATypeVariable(subTypeSym.type)));
-                    }
-                    if (subTypeSym.isAnonymous() && !c.isEnum()) {
-                        log.error(TreeInfo.diagnosticPositionFor(subTypeSym, env.tree),  Errors.LocalClassesCantExtendSealed(Fragments.Anonymous));
-                    }
-                    if (permittedTypes.contains(subTypeSym)) {
-                        DiagnosticPosition pos =
-                                env.enclClass.permitting.stream()
-                                        .filter(permittedExpr -> TreeInfo.diagnosticPositionFor(subTypeSym, permittedExpr, true) != null)
-                                        .limit(2).collect(List.collector()).get(1);
-                        log.error(pos, Errors.InvalidPermitsClause(Fragments.IsDuplicated(subTypeSym.type)));
-                    } else {
-                        permittedTypes.add(subTypeSym);
-                    }
-                    if (sealedInUnnamed) {
-                        if (subTypeSym.packge() != c.packge()) {
+            // The info.lint field in the envs stored in typeEnvs is deliberately uninitialized,
+            // because the annotations were not available at the time the env was created. Therefore,
+            // we look up the environment chain for the first enclosing environment for which the
+            // lint value is set. Typically, this is the parent env, but might be further if there
+            // are any envs created as a result of TypeParameter nodes.
+            Env<AttrContext> lintEnv = env;
+            while (lintEnv.info.lint == null)
+                lintEnv = lintEnv.next;
+
+            // Having found the enclosing lint value, we can initialize the lint value for this class
+            env.info.lint = lintEnv.info.lint.augment(c);
+
+            Lint prevLint = chk.setLint(env.info.lint);
+            JavaFileObject prev = log.useSource(c.sourcefile);
+            ResultInfo prevReturnRes = env.info.returnResult;
+
+            try {
```

Comment:
Previously this try block didn't include this code. Is, its new position intentional?